test/prefer-modern-dom-apis.js

```diff
@@ -0,0 +1,129 @@
+		{
+			code: 'parentNode.replaceChild(newChildNode, oldChildNode);',
+			errors: [
+				{
+					message:
+						'Prefer `oldChildNode.replaceWith(newChildNode)` over `parentNode.replaceChild(newChildNode, oldChildNode)`.'
+				}
+			],
+			output: 'oldChildNode.replaceWith(newChildNode);'
+		},
+		// Tests for .insertBefore()
+		{
+			code: 'parentNode.insertBefore(newNode, referenceNode);',
+			errors: [
+				{
+					message:
+						'Prefer `referenceNode.before(newNode)` over `parentNode.insertBefore(newNode, referenceNode)`.'
+				}
+			],
+			output: 'referenceNode.before(newNode);'
+		},
+		// Tests for .insertAdjacentText()
+		{
+			code: 'referenceNode.insertAdjacentText("beforebegin", "text");',
+			errors: [
+				{
+					message:
+						'Prefer `referenceNode.before("text")` over `referenceNode.insertAdjacentText("beforebegin", "text")`.'
+				}
+			],
+			output: 'referenceNode.before("text");'
+		},
+		{
+			code: 'referenceNode.insertAdjacentText("afterbegin", "text");',
+			errors: [
+				{
+					message:
+						'Prefer `referenceNode.prepend("text")` over `referenceNode.insertAdjacentText("afterbegin", "text")`.'
+				}
+			],
+			output: 'referenceNode.prepend("text");'
+		},
+		{
+			code: 'referenceNode.insertAdjacentText("beforeend", "text");',
+			errors: [
+				{
+					message:
+						'Prefer `referenceNode.append("text")` over `referenceNode.insertAdjacentText("beforeend", "text")`.'
+				}
+			],
+			output: 'referenceNode.append("text");'
+		},
+		{
+			code: 'referenceNode.insertAdjacentText("afterend", "text");',
+			errors: [
+				{
+					message:
+						'Prefer `referenceNode.after("text")` over `referenceNode.insertAdjacentText("afterend", "text")`.'
+				}
+			],
+			output: 'referenceNode.after("text");'
+		},
+		// Tests for .insertAdjacentElement()
+		{
+			code: 'referenceNode.insertAdjacentElement("beforebegin", newNode);',
+			errors: [
+				{
+					message:
+						'Prefer `referenceNode.before(newNode)` over `referenceNode.insertAdjacentElement("beforebegin", newNode)`.'
+				}
+			],
+			output: 'referenceNode.before(newNode);'
+		},
+		{
+			code: 'referenceNode.insertAdjacentElement("afterbegin", "text");',
+			errors: [
+				{
+					message:
+						'Prefer `referenceNode.prepend("text")` over `referenceNode.insertAdjacentElement("afterbegin", "text")`.'
+				}
+			],
+			output: 'referenceNode.prepend("text");'
+		},
+		{
+			code: 'referenceNode.insertAdjacentElement("beforeend", "text");',
+			errors: [
+				{
+					message:
+						'Prefer `referenceNode.append("text")` over `referenceNode.insertAdjacentElement("beforeend", "text")`.'
+				}
+			],
+			output: 'referenceNode.append("text");'
+		},
+		{
+			code: 'referenceNode.insertAdjacentElement("afterend", newNode);',
+			errors: [
+				{
+					message:
+						'Prefer `referenceNode.after(newNode)` over `referenceNode.insertAdjacentElement("afterend", newNode)`.'
+				}
+			],
+			output: 'referenceNode.after(newNode);'
+		}
+	]
+});
```

Comment:
Add tests for the following scenarios:

```
referenceNode.insertAdjacentElement(foo, bar);

referenceNode.insertAdjacentElement(
    "afterend",
    newNode
);

referenceNode.insertAdjacentElement( // inline comments
    "afterend", // inline comments
    newNode,  // inline comments
); // inline comments

parentNode.replaceChild(
    newChildNode,
    oldChildNode
);

parentNode.replaceChild( // inline comments
    newChildNode, // inline comments
    oldChildNode // inline comments
);

parentNode.insertBefore(alfa, beta).insertBefore(charlie, delta);
```

Reply:
Cool, thanks for these test scenario suggestions!
So I guess the inline comments should be preserved? How can I do that?

Reply:
It's a total pain. If you want, you can disable the fixer if there are comments in the node. Take a look at `getCommentsInside` or `commentsExistBetween`.

https://eslint.org/docs/developer-guide/working-with-rules#contextgetsourcecode

Reply:
`referenceNode.insertAdjacentElement(foo, bar);` -> this is incorrect syntax, I would not touch it and not modify it and add it to the valid array, correct?

`parentNode.insertBefore(alfa, beta).insertBefore(charlie, delta);` -> how to handle this?

Reply:
`referenceNode.insertAdjacentElement(foo, bar);` should be just as valid as `referenceNode.insertAdjacentElement("beforebegin", newNode);`. It's just using a variable instead of a literal.

----

`parentNode.insertBefore(alfa, beta).insertBefore(charlie, delta);` cannot be fixed properly because the return value is being used. A similar test would be `var foo = parentNode.insertBefore(alfa, beta);`